Sources/CacheAPI-postgres.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines http://www.simplemachines.org
+ * @copyright 2017 Simple Machines and individual contributors
+ * @license http://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 2.1 Beta 3
+ */
+
+if (!defined('SMF'))
+	die('Hacking attempt...');
+
+/**
+ * PostgreSQL Cache API class
+ * @package cacheAPI
+ */
+class postgres_cache extends cache_api
+{
+
+	public function __construct()
+	{
+		parent::__construct();
+
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function connect()
+	{
+		global $db_prefix, $db_connection;
+
+		pg_prepare($db_connection, '', 'SELECT 1 
+			FROM   pg_tables
+			WHERE  schemaname = $1
+			AND    tablename = $2');
+
+		$result = pg_execute($db_connection, '', array('public', $db_prefix . 'cache'));
+
+		if(pg_affected_rows($result) === 0)
+			pg_query($db_connection, 'CREATE TABLE {db_prefix}cache (key text, value text, ttl bigint, PRIMARY KEY (key))');			
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function isSupported($test = false)
+	{
+		global $smcFunc, $db_connection;
+		
+
+		if ($smcFunc['db_title'] !== 'PostgreSQL')
+			return false;
+
+		$result = pg_query($db_connection, 'SHOW server_version_num');
+		$res = pg_fetch_assoc($result);
+		
+		if($res['server_version_num'] < 90500)
+			return false;
+		
+		return true;
+	}
```

Comment:
Don't forget "$test = true" is what tests if we can use the API, and without it we try to actually use it.  It should fail after this if the parent doesn't show support.  The parent fails isSupported if caching is disabled.  For your code, I think the easiest is to just replace the return true with return parent::isSupported();